storage/store/inmemory/unsynchronized/light_transaction_results.go

```diff
@@ -110,3 +111,25 @@ func (l *LightTransactionResults) BatchStore(flow.Identifier, []flow.LightTransa
 func (l *LightTransactionResults) BatchStoreBadger(flow.Identifier, []flow.LightTransactionResult, storage.BatchStorage) error {
 	return fmt.Errorf("not implemented")
 }
+
+// AddToBatch adds all the in-memory storages to the given batch.
+// It is used for the batching writes to the DB.
+func (l *LightTransactionResults) AddToBatch(batch storage.ReaderBatchWriter) error {
+	writer := batch.Writer()
+
+	for block, results := range l.blockStore {
+		decodedBlock, err := store.KeyToBlockID(block)
+		if err != nil {
+			return fmt.Errorf("could not decode block: %w", err)
+		}
+
+		for _, txResult := range results {
+			err = operation.BatchInsertLightTransactionResult(writer, decodedBlock, &txResult)
+			if err != nil {
+				return fmt.Errorf("could not persist light transaction result: %w", err)
+			}
+		}
```

Comment:
this also needs to index the results
https://github.com/onflow/flow-go/blob/1186df0d19b5f1b632f790a0f6226c7f4f68be5e/storage/store/light_transaction_results.go#L86-L96